sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -84,15 +84,33 @@ export function parseHealthcareEntityIndex(pointer: string): number {
 
 const jsEncodingUnit = "Utf16CodeUnit";
 
-export function addStrEncodingParam<T>(options: T): T & { stringIndexType: StringIndexType } {
-  return { ...options, stringIndexType: jsEncodingUnit };
+/**
+ * Measurement units that can used to calculate the offset and length properties.
+ */
+export type StringIndexType = "TextElements_v8" | "UnicodeCodePoint" | "Utf16CodeUnit";
+
+export function addStrEncodingParam<T>(
+  options: T & { stringIndexType?: StringIndexType }
+): T & { stringIndexType: StringIndexType } {
+  return { ...options, stringIndexType: options.stringIndexType || jsEncodingUnit };
+}
```

Comment:
TypeScript isn't actually complex enough yet to infer `T` to be the instantiated type of the parameter with stringIndexType omitted, it is inferred to be the whole type of the parameter. Slightly more natural and TypeScript-y IMO:

```suggestion
export function addStrEncodingParam<Options extends { stringIndexType?: StringIndexType }>(
  options: Options
): Options & { stringIndexType: StringIndexType } {
  return { ...options, stringIndexType: options.stringIndexType || jsEncodingUnit };
}
```

Reply:
Fixed